lib/TreeSearch.js

```diff
@@ -5,50 +5,57 @@ const TreeSearch = {};
 // Returns a function that performs a preorder search on the tree for the given
 // simplification function
 TreeSearch.preOrder = function(simplificationFunction) {
-  return function (node) {
-    return search(simplificationFunction, node, true);
+  return function (node, options={}) {
+    return search(simplificationFunction, node, true, options);
   };
 };
 
 // Returns a function that performs a postorder search on the tree for the given
 // simplification function
 TreeSearch.postOrder = function(simplificationFunction) {
-  return function (node) {
-    return search(simplificationFunction, node, false);
+  return function (node, options={}) {
+    return search(simplificationFunction, node, false, options);
   };
 };
 
 // A helper function for performing a tree search with a function
-function search(simplificationFunction, node, preOrder) {
+function search(simplificationFunction, node, preOrder, options={}) {
   let status;
 
   if (preOrder) {
-    status = simplificationFunction(node);
+    status = simplificationFunction(node, options);
     if (status.hasChanged()) {
       return status;
     }
   }
 
-  if (Node.Type.isConstant(node) || Node.Type.isSymbol(node)) {
+  if (Node.Type.isConstant(node)) {
     return Node.Status.noChange(node);
   }
+  // Check isUnaryMinus before Symbol because symbols nested inside unaryMinus
+  // (e.g., 2x - symbol) need to be subtracted from the expression. Checking
+  // symbol first causes the symbol to be added.
```

Comment:
I think with my unary minus change, it doesn't have to be checked first - can you switch it back and see if that works? Before, symbols with unary minuses were counted as symbols too by default, but now it `Node.Type.isSymbol(node)` on a unary minus node should return false by default :D